libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -755,6 +810,24 @@ BeaconSetupService::State BeaconSetupService::OnDryRun()
 BeaconSetupService::State BeaconSetupService::OnBeaconReady()
 {
   FETCH_LOCK(mutex_);
+
+  // Populate beacon entropy with notarisation keys of qual
+  if (notarisation_callback_function_)
+  {
+    std::map<MuddleAddress, NotarisationManager::PublicKey> qual_notarisation_keys;
+    for (auto const &member : beacon_->manager.qual())
+    {
+      assert(notarisation_key_msgs_.find(member) != notarisation_key_msgs_.end());
+      auto notarisation_msg = notarisation_key_msgs_.at(member);
+      beacon_->block_entropy.aeon_notarisation_keys.insert(
+          {member, std::make_pair(notarisation_msg.PublicKey(), notarisation_msg.Signature())});
```

Comment:
No, the assert is there for debugging. The change to include valid_dkg_members is so that you do not process any messages from people who have not sent you notarisation keys so they should never be part of your qual. If someone doesn't populate with notarisations or with incorrect ones then the aeon beginning block produced by that person is just invalid. As long as there is one honest member in qual then there will be a aeon beginning block.